### PR TITLE
Change mark_cache size calculation to cut down memory usage

### PR DESCRIPTION
**Pull request by hongyunyan**
### What problem does this PR solve?

Issue Number: ref #6426

Problem Summary:

### What is changed and how it works?

Currently, we calculate the mark_cache memory usage with marks.size() * sizeof(MarkInCompressedFile);
However, PODArray will do roundUpToPowerOfTwoOrZero when allocate element memory usage.
Thus, the memory usage is quite bigger then the calculation ways, which is close in twice time of the calculation ways.
To ensure the mark_cache only use the memory as we allocate, we change the memory usage calculation approach.

Then, we take the memory of allocation for MarkInCompressedFile for consideration. I think when we just initialization MarkInCompressedFile with the element number, we don't need to make the memory roundUpToPowerOfTwoOrZero. Thus, we change the alloc in  PODArray(size n), to cut down the usage.

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No code

I compare the cluster with master tiflash and tiflash of this PR, to run a typical Workload.
We can find for the master tiflash, the memory will reach 13.0GB steeply and then continue slowing growth. 
While for the tilfash of this PR, the memory will reach 11.5GB steeply and then continue slowing growth.
Thus, in a typical workload, we can cut down nearly 1.5GB.

image
image

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
Reduce the resident memory usage of mark_cache.
```

### CODE REVIEW

File: dbms/src/Common/PODArray.h [hunk 1]
```diff
@@ -116,7 +116,7 @@ class PODArrayBase : private boost::noncopyable
 
     void alloc_for_num_elements(size_t num_elements)
     {
-        alloc(roundUpToPowerOfTwoOrZero(minimum_memory_for_elements(num_elements)));
+        alloc(minimum_memory_for_elements(num_elements));
```
Comment (flowbehappy) [inline dbms/src/Common/PODArray.h:119]: Plz add some comments to this method. About why don't we use roundUpToPowerOfTwoOrZero.

**Review by breezewish — APPROVED** [review]
The reduce of memory seems to be not as notable as expected. But anyway it's an improvement. Congrats!

**Pull request merged**